### DISCUSSION TITLE
Turning off NOZZLE_CLEAR macro

### DISCUSSION THREAD

**Discussion by kpipes556** [Guilouz/Creality-Helper-Script-Wiki#609] [General]
Any idea how to turn this macro off? It ends up making the nozzle dirtier because it oozes during the heating and doesn't fully clean it off.

I commented out CX_NOZZLE_CLEAR and it's still doing it.

**Comment by TheWebMachine** [comment]
Where did you comment it out from? Should be the START_PRINT macro.

**Reply by kpipes556** [reply]
Yep, I commented it out in the new START_PRINT macro in the KAMP folder, right between the two G28s, and it did it anyway like I had never commented it out.

**Reply by TheWebMachine** [reply]
Oh, I don't use KAMP. Perhaps you need to comment it out in the original START_PRINT macro just in case it is somehow referencing that one. When I comment it out there, it works. Alternatively, you could probably trip the ready flag manually, perform your own home, tilt check, and/or mesh, then fire off START_PRINT in your gcode with it thinking *it* already did the tilt and nozzle cleaning.

Since I don't use KAMP (yet? lol), I could be missing something else. 🤷🏻‍♂️

**Reply by kpipes556** [reply]
Wait that's useful info though! I don't really need KAMP... I can just root and SSH into the stock START_PRINT and comment that out and it works for you? No nozzle cleaning routine?

I thought I had tried that before and it didn't work. Maybe I'm doing it wrong somehow...

**Reply by TheWebMachine** [reply]
It did work for me that way, yeah. However, I don't think I did it on 1.3.3.8 (I'm using nozzle clean rn)...I last tried it on 1.3.3.5 back before I transplanted the brush to the standalone carrier so I could use it with other plates. Didn't try it again when I upgraded to .8 and set everything back up.

I haven't upgraded to the latest .29 version yet, either. I have a K1C, if that at all matters in this case.

**Comment by TheWebMachine** [comment]
I only found nozzle crud after cleaning to be a problem with sticky PETGs. Are you having the issue with PLA, ABS, etc?

**Reply by kpipes556** [reply]
TPU is the worst offender, though it has happened with PLA and PETG as well. I think it calculates the NOZZLE_CLEAR extruder temp to be your set temp - 20 and leaves it there. For a lot of filaments, 20 below operating temp is plenty enough for it to ooze. I have a K1C also (though I honestly am so unimpressed with Creality's customer service I might own a Bambu soon...)

**Reply by TheWebMachine** [reply]
Really? The CS was actually really good to me when my heater core mount turned out to be defective (captives sheered right out of the bores). Quickly sent me a replacement and answered all my questions pretty fast. I was actually really surprised at how they were on that ticket. Must really boil down to who your ticket gets assigned to. 🤷🏻‍♂️

**Reply by TheWebMachine** [reply]
And yes, the nozzle clean and the nozzle design itself could use some work. 

There is an STL out there for a nozzle mold to pour your own replacements. I might modify it to give it slightly larger and longer bristles to clean further up the nozzle head *and* add a solid wiper to either side to fling away debris.

**Reply by kpipes556** [reply]
Huh, I have contacted them >3 times with this issue and they never once followed up with even an email, they would say it's been forwarded to R&D and haven't heard anything back. That's from trying their chat service, their email, and their customer support Facebook chat

**Comment by adoyle-h** [comment]
Has anyone found how to turn off NOZZLE_CLEAR?

**Comment by j5immortal** [comment]
I eventually figured out a method to prevent the NOZZLE_CLEAR.

It works whether you are using KAMP or not.  You will have to do this by running a terminal like MobaXterm. 
Start the SSH session and login to the printer to access your printer's root files. 
Click on the green "UP Arrow" parent directory icon to see the root directory files. 
Select - usr - share - klipper - klippy - extras - **custom_macro.py**
Comment the following two lines out with "#" where indicated  in the code blocks  indicated below:
  
 cmd_CX_ROUGH_G28_help = "rough G28"
 . . .
      # self.gcode.run_script_from_command('NOZZLE_CLEAR HOT_MIN_TEMP=%d HOT_MAX_TEMP=%d BED_MAX_TEMP=%d' % (self.g28_ext_temp, self.extruder_temp - 20, self.bed_temp))

 cmd_CX_NOZZLE_CLEAR_help = "nozzle clear with temperature"
 . . .
       # self.gcode.run_script_from_command('NOZZLE_CLEAR HOT_MIN_TEMP=%d HOT_MAX_TEMP=%d BED_MAX_TEMP=%d' % (self.g28_ext_temp, self.extruder_temp - 20, self.bed_temp))
        
 Restart your printer

**Reply by adoyle-h** [reply]
Awesome! That's what I was looking for. Thank you.

**Reply by j5immortal** [reply]
Pleasure, please note that skipping the nozzle clear sequence may influence the nozzle height calibration before each print if plastic is sticking to the nozzle.  You will have to manually ensure the nozzle is clean

**Reply by UglyBird1** [reply]
Hallo j5immortal, ich nutze version 1.3.3.46 und es funktioniert super. Wenn ich vorher PLA_LW nutzte, musste ich teiweise 20 Minuten warten. Das hat jetzt; dank dir; ein Ende.
Für die Unwissenden, wie ich, in der Version MobaXterm v26.4 ist der Ordner zum öffnen der Datei kein grüner Pfeil, sondern ein Ordner Ikon mit schwarzem Pfeil. Und nicht vergessen alles zu speichern!!!!!!!!!!! Dann Restart des Druckers!!!!!!!!

**Comment by tpruvot** [comment]
Here is how i replaced it in printer.cfg : 
```
# Include this to avoid K1Max bed damages on print startup (tpruvot Dec. 2024)

[gcode_macro CX_NOZZLE_CLEAR]
rename_existing: cx_nozzle_clear_py
gcode:
  G0 X20 Y300 Z4 F4000
  G0 X130 Y303 Z3 F4000
  G0 X180 Y303 Z3 F2000
  G0 X130 Y303 Z3 F2000
  G0 X180 Y303 Z3 F4000
  G0 X130 Y303 Z3 F2000
  G0 X180 Y303 Z3 F4000

[gcode_macro NOZZLE_CLEAR]
rename_existing: nozzle_clear_py
gcode:
  # do nothing
```

Look like you need both overrides, eventually you can copy paste the CX one in the second one to clean after the purge line if you add one of these silicon wipe stickers (1€ on aliexpress). Else you can just comment the G0 lines to fully avoid the wipe stuff. I need to recheck the second macro call with the 1.3.3.46, look like they changed something after the line. Was supposed to be cleaned there ... but its maybe only done with Z calibration (G29)

Out of subject my other change in these macros is in the START_PRINT macro (gcode_macro.cfg), i've set quiet mode by default and a small retraction after the purge line to avoid/reduce tracks on the bed: 
```

[gcode_macro START_PRINT]
  ...
  Qmode
  CX_PRINT_DRAW_ONE_LINE
  {% if printer.extruder.can_extrude|lower == 'true' %}
    G1 E-1.0 F180
  {% endif %}
```

**Reply by krispi69** [reply]
Another method to disable Nozzle Clean and gain access to the file start_print.cfg:

In the web interface file explorer, go to Configuration → Helper-Script → KAMP.

Create a new file named Start_Print_TEMP.cfg.

Copy the contents of Start_Print.cfg and paste them into the newly created Start_Print_TEMP.cfg file.

Find the line containing CX_NOZZLE_CLEAR → add a “#” before it → #CX_NOZZLE_CLEAR.

Save the file.

Open MobaXterm, PuTTY, or any other SSH terminal.

Connect to the printer’s IP address using user: root and password: creality_2023.

Run the command:

cd /usr/data/printer_data/config/Helper-Script/KAMP


Check if the files exist:

ls -l | grep Start_Print


You should see both files listed with their permissions.

Rename the original file:

mv Start_Print.cfg Start_Print_BACKUP.cfg


Rename the temporary file to replace it:

mv Start_Print_TEMP.cfg Start_Print.cfg


Finally, it’s best to restart the Klipper host.
